haralyzer/assets.py

```diff
@@ -324,6 +333,14 @@ def filter_entries(self, request_type=None, content_type=None,
             if http_version is not None and not p.match_http_version(
                     entry, http_version, regex=regex):
                 valid_entry = False
+            if source is not None and source == 'internal':
+                if p.hostname != get_entry_hostname(entry):
+                    valid_entry = False
+            elif source is not None and source == 'external':
+                if p.hostname == get_entry_hostname(entry):
+                    valid_entry = False
+            elif source is not None and isinstance(source, str) == True:
```

Comment:
The `isinstance` check seems redundant here, since we already know the input is invalid (i.e. - the keyword argument is not None, and is not either of the things we want/expect).